src/main/java/pages/GameGenrePage.java

```diff
@@ -0,0 +1,194 @@
+        try {
+            return Utils.replaceStringWithoutTradeMark(gameItem.findElement(gamesNameBy).getText());
+        } catch (Error e) {
+            var errMes = "Couldn't get game name. \nDetailed message: \n" + e.getMessage();
+            log.error(errMes);
+            throw new Exception(errMes);
+        }
+    }
+
+    @SneakyThrows
+    private Double getFinalPrice(WebElement gameItem, By gameFinalPriceLocator) {
+        try {
+            if (gameItem.findElements(gameFinalPriceLocator).size() == 0) {
+                return 0.0;
+            }
+            String gamePrice = gameItem.findElement(gameFinalPriceLocator).getText();
+            if (gamePrice.contains("Free")) {
+                return 0.0;
+            } else {
+                return Utils.getNumberFormString(gamePrice);
+            }
+        } catch (Error e) {
+            var errMes = "Couldn't get final price. \nDetailed message: \n" + e.getMessage();
+            log.error(errMes);
+            throw new Exception(errMes);
+        }
+    }
+
+    @SneakyThrows
+    public static Double getFinalPrice(WebElement gameFinalPriceLocator) {
+        try {
+            String gamePrice = gameFinalPriceLocator.getText();
+            if (gamePrice.contains("Free") || gamePrice.equals("")) {
+                return 0.0;
+            } else {
+                return Utils.getNumberFormString(gamePrice);
+            }
+        } catch (Error e) {
+            var errMes = "Couldn't get final price. \nDetailed message: \n" + e.getMessage();
+            log.error(errMes);
+            throw new Exception(errMes);
+        }
+    }
+
+    @SneakyThrows
+    private Double getPriceWithDiscountIfExists(WebElement gameItem) {
+        try {
+            if (gameItem.findElements(gameOriginalPriceBy).size() > 0) {
+                return Utils.getNumberFormString(gameItem.findElement(gameOriginalPriceBy).getText());
+            } else {
+                return null;
+            }
+        } catch (Error e) {
+            var errMes = "Couldn't get price with discount. \nDetailed message: \n" + e.getMessage();
+            log.error(errMes);
+            throw new Exception(errMes);
+        }
+    }
+
+    @SneakyThrows
+    private Double getDiscountIfExists(WebElement gameItem) {
+        try {
+            if (gameItem.findElements(gameDiscountBy).size() > 0) {
+                return Utils.getNumberFormString(gameItem.findElement(gameDiscountBy).getText());
+            } else {
+                return 0.0;
+            }
+        } catch (Error e) {
+            var errMes = "Couldn't find discount. \nDetailed message: \n" + e.getMessage();
+            log.error(errMes);
+            throw new Exception(errMes);
+        }
+    }
+
+    @SneakyThrows
+    private List<GenreCatalogGameItem.Platform> getPlatforms(WebElement gameItem) {
+        try {
+            List<GenreCatalogGameItem.Platform> platformList = new ArrayList<>();
```

Comment:
I think you can find more universal locator and don't use a lot of if conditions